packages/app/components/Questions/Queue/TaskMarkingSelector.tsx

```diff
@@ -0,0 +1,158 @@
+import { ReactElement, useState } from 'react'
+import { Text } from '../Shared/SharedComponents'
+import { ConfigTasks } from '@koh/common'
+
+/*
+  This component is used when the TA is helping a student (used on the student queue card). 
+  Allows the TA to choose what tasks (of the one's the student is asking to be marked) are good and should be marked 
+*/
+
+interface CheckableMarkingTaskProps {
+  taskId: string
+  taskName: string
+  taskColor: string
+  onChange: (taskId: string, checked: boolean) => void
+  checked: boolean
+  decorative: boolean
+}
+
+function CheckableMarkingTask({
+  taskId,
+  taskName,
+  taskColor,
+  onChange,
+  checked,
+  decorative,
+}: CheckableMarkingTaskProps): React.ReactElement {
+  function getBrightness(color: string): number {
+    const rgb = parseInt(color.slice(1), 16)
+    const r = (rgb >> 16) & 0xff
+    const g = (rgb >> 8) & 0xff
+    const b = (rgb >> 0) & 0xff
+    return (r * 299 + g * 587 + b * 114) / 1000
+  }
+  const textColor = checked
+    ? getBrightness(taskColor) < 128
+      ? 'white'
+      : 'black'
+    : 'gray'
+
+  const handleClick = () => {
+    onChange(taskId, !checked)
+  }
+
+  // for making it so you can press enter to toggle it
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    if (event.key === 'Enter') {
+      handleClick()
+    }
+  }
+
+  // for applying hover and focus styles
+  const [isHovered, setIsHovered] = useState(false)
+  const handleMouseEnter = () => {
```

Comment:
We have 4 methods that call same useState, should we reduce amount of methods and make it more general?

Reply:
it is already more general. If someone wanted to change a state differently for focus/blur compared to mouseEnter/mouseLeave, they can. Also, the function's names are accurate to what they are doing (handling an action).  It could be probably changed to 2 tiny functions instead of 4, but meh, I would argue it does not matter much in this case